### PR TITLE
chore: Update ng-select to support Screen Readers

### LINKED ISSUE #10887: Update ng-select to support Screen Readers
- This ticket contains improvements for [Headings and Labels Success Criterion](https://github.com/SAP/spartacus/issues/11192)  ([WCAG 2.1 2.4.6](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html))
- master ticket: #11192

`ng-select` is a generic component shared across the application and has limited a11y support. There are workaround to make it more a11y friendly, but it may not be worth it and cleaner solutions could be to switch all `ng-select` to `native select`. 

**Screen Reader issue:** ng-select is announced as a list box, but the default value or place holder is not announced, in this case, "Relevance". ng-select should support `aria-label attribute`

## Improvements 
- [x] Investigate workaround to add support for `aria-label`.

**Solution:** Upgrade ng-select to v5.10.0 which offers `aria-lable` support

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #10887

### CODE REVIEW

File: feature-libs/organization/administration/components/shared/list/list.component.html [hunk 1]
```diff
@@ -21,7 +21,7 @@ <h3>
           </h3>
         </div>
 
-        <div class="actions">
```
Comment (Pucek9) [inline feature-libs/organization/administration/components/shared/list/list.component.html:24]: test

**Comment by Pucek9** [comment]
QA steps:
find all occurrences of ng-select and verify that they have label. Final result should be that every inputs are read by SR like "label value"

b2c:
- product sort: any user, open any product list page
- payment expiration month & year: any customer, go through checkout
- myCoupons: test-user-with-coupons@sap.cx.com / pw4all 

- orderHistory: any user, do some orders and check order history
- myInterests: test-user-with-orders@sap.cx.com / pw4all - (set notification preferences and add one for
`Secure Digital Card 2GB` if list empty)
- returnRequestList: - not sure how to test

b2b & powertools
- organziation sort list: linda.wolf@rustic-hw.com / pw4all - open budgets list or other feature, it should be empty for units)
- replenishment orders: alejandro.navarro@rustic-hw.com / pw4all (do checkout with replenishment if list empty)

**Comment by saad-mo** [comment]
My QA checklist 

b2c:

- [x] product sort: any user, open any product list page

- [x] payment expiration month & year: any customer, go through checkout

- [x] myCoupons: test-user-with-coupons@sap.cx.com / pw4all

- [x] order history: any user, do some orders and check order history

- [x]  myInterests: test-user-with-orders@sap.cx.com / pw4all - (set notification preferences and add one for
Secure Digital Card 2GB if list empty)

- [ ]  returnRequestList: - not sure how to test

b2b & powertools

- [ ]  organziation sort list: linda.wolf@rustic-hw.com / pw4all - open budgets list or other feature, it should be empty for units)
- [ ]  replenishment orders: alejandro.navarro@rustic-hw.com / pw4all (do checkout with replenishment if list empty)

File: projects/storefrontlib/src/cms-components/myaccount/order/return-request-list/order-return-request-list.component.html [hunk 2]
```diff
@@ -5,19 +5,16 @@
...
               [sortOptions]="returnRequests.sorts"
               [sortLabels]="getSortLabels() | async"
               (sortListEvent)="changeSortCode($event)"
               [selectedOption]="returnRequests.pagination.sort"
-              placeholder="{{
-                'returnRequestList.sortByMostRecent' | cxTranslate
-              }}"
             ></cx-sorting>
-          </div>
+          </label>
```
Comment (saad-mo) [inline projects/storefrontlib/src/cms-components/myaccount/order/return-request-list/order-return-request-list.component.html:17]: We need this label for the bottom select as well

File: feature-libs/organization/administration/components/shared/list/list.component.html [hunk 3]
```diff
@@ -22,19 +22,25 @@ <h3>
...
 
         <div class="actions">
-          <ng-select
-            class="sort"
-            *ngIf="data.pagination?.sort"
-            [searchable]="false"
-            [clearable]="false"
-            (change)="sort(data.pagination)"
-            [tabIndex]="0"
-            [(ngModel)]="sortCode"
```
Comment (saad-mo) [inline feature-libs/organization/administration/components/shared/list/list.component.html:32]: missing `[attr.aria-label]="selectedLabel || placeholder"` to match with `ng-select` used in `cx-sorting` component

File: projects/storefrontlib/src/cms-components/myaccount/order/return-request-list/order-return-request-list.component.html (hunk 2 above)
Comment (saad-mo) [reply projects/storefrontlib/src/cms-components/myaccount/order/return-request-list/order-return-request-list.component.html:17]: tested and fixed

**Review by saad-mo — APPROVED** [review]
Looks good to me 👍

**Pull request merged**